svc.go

```diff
@@ -169,16 +210,19 @@ func (as *AutoNATService) doDial(pi peer.AddrInfo) *pb.Message_DialResponse {
 	// rate limit check
 	as.mx.Lock()
 	count := as.reqs[pi.ID]
-	if count >= AutoNATServiceThrottle {
+	if count >= AutoNATServiceThrottle || as.globalReqs >= AutoNATGlobalThrottle {
```

Comment:
Hm. We should _probably_ be checking these _before_ we do any work but we can move them later.